### PR TITLE
[ZIP-14] Add zip for revised pBFT consensus

### PR DESCRIPTION
**Pull request by sandipbhoir**

### CODE REVIEW

File: zips/zip-14.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+### Old pBFT
+Please refer below for more details on old behaviour of pBFT for shard microblock.
+
+![existing_flow](../assets/zip-14/existing_flow.png)
+
+### Revised PBFT
+Please refer below for more details on new behaviour of pBFT for shard microblock.
+Same applies for finalblock consensus.
+
+![new_flow](../assets/zip-14/new_flow.png)
```
Comment (bb111189) [inline zips/zip-14.md:25]:
```suggestion
![new_flow](../assets/zip-14/revised_new_flow.png )
```

File: zips/zip-14.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+|  ZIP | Title | Status| Type | Author | Created (yyyy-mm-dd) | Updated (yyyy-mm-dd)
+|--|--|--|--| -- | -- | -- |
+| 14  | Revised pBFT consensus | Draft | Standards Track  | Sandip Bhoir <sandip@zilliqa.com>| 2020-12-10 | 2020-12-10
```
Comment (bb111189) [inline zips/zip-14.md:3]: Let's add George Pîrlea george@zilliqa.com to the ZIP

File: zips/zip-14.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+### Revised PBFT
+Please refer below for more details on new behaviour of pBFT for shard microblock.
+Same applies for finalblock consensus.
+
+![new_flow](../assets/zip-14/revised_new_flow.png )
+
+These are important points to consider:
+
+**Round 1:**
+- Round 1 of consensus also called preprep phase will run consensus on microblock/finalblock with txnhashes only.
```
Comment (bb111189) [inline zips/zip-14.md:30]:
```suggestion
- Round 1 of consensus also known as the `pre-prepare` phase will run consensus on microblock/finalblock with a set of transaction hashes. This phase is for the committee to agree on the order and set of transactions to process. 
```

File: zips/zip-14.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+- Leader starts actual transaction processing after sending out preprep announcement ( mb/fb with txnhashes only).
```
Comment (bb111189) [inline zips/zip-14.md:31]:
```suggestion
- Leader starts transaction exeuction after sending out `pre-prepare` announcement ( mb/fb with transaction hashes only).
```

File: zips/zip-14.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+- Backups starts actual transaction processing after sending out commit for preprep phase.
```
Comment (bb111189) [inline zips/zip-14.md:32]:
```suggestion
- Backups start transaction execution after sending out commit to the leader for `pre-prepare` phase.
```

File: zips/zip-14.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+- However, if there is commit failure ( ex. due to missing txns), it will fetch the missing ones from leader and rerun preprep phase (round-1) of consensus.
```
Comment (bb111189) [inline zips/zip-14.md:33]:
```suggestion
- However, if there is commit failure ( ex. due to missing txns), the backuip will fetch the missing transactions from leader and rerun the `pre-prepare` phase (round-1) of consensus.
```

**Review by bb111189 — CHANGES_REQUESTED** [review]
see comments

**Review by ansnunez — APPROVED** [review]

**Review by bb111189 — APPROVED** [review]

**Pull request merged**